### ISSUE TITLE
It's not friendly for small cpu memory users to get fid.

### ISSUE THREAD

**Issue by Feywell** [mit-han-lab/gan-compression#44]
I try to get my own dataset by  `get_real_stat.py` .
But I get a error need more than 13G memory in  `tensors = util.tensor2im(tensors).astype(float)`
So, I try to reduce memory.
Firstly, It is simple to change it to  `tensors = util.tensor2im(tensors).astype(np.float32)`. It is useful.
But not good.
I try to change function `get_activations_from_ims` in `fid_score.py` like:
```
            images = images.transpose((0, 3, 1, 2))
        images = images.astype(np.float32)/255
```
But I  find it is not good enought by memory_profiler, like follow:
```
Line #    Mem usage    Increment   Line Contents
================================================
    10   9041.7 MiB   9041.7 MiB   @profile
    11                             def get_fid(fakes, model, npz, device, batch_size=1, use_tqdm=True, bgr=False):
    12   9041.7 MiB      0.0 MiB       m1, s1 = npz['mu'], npz['sigma']
    13  15747.4 MiB   6705.7 MiB       fakes = torch.cat(fakes, dim=0)
    14  10226.1 MiB      0.0 MiB       fakes = util.tensor2im(fakes, normalize=False)  #.astype(np.float32)   # default float
    15  10226.1 MiB      0.0 MiB       m2, s2 = _compute_statistics_of_ims(fakes, model, batch_size, 2048,
    16  10338.3 MiB    112.3 MiB                                           device, use_tqdm=use_tqdm, bgr=bgr)
    17  10346.7 MiB      8.4 MiB       return float(calculate_frechet_distance(m1, s1, m2, s2))
```

Do you have some good ideas to reduce memory?
Thank you!

**Comment by lmxyy** [comment]
I think a more memory-efficient for `get_real_stat.py` is to store the image tensor on the disk and then modify the [dataloader](https://github.com/mit-han-lab/gan-compression/blob/03c30b20de9094a31cb7bfbb5243134a6e605132/metric/fid_score.py#L189) to restore the tensors from the disk. In this way, your memory issue could be addressed.

**Comment by Feywell** [comment]
Thank you！It is a good iead

**Comment by deepseek** [comment]
@lmxyy can you explain more

> store the image tensor on the disk and then modify the dataloader to restore the tensors from the disk
